modules/runner/src/integrationTest/java/org/apache/ignite/internal/runner/app/client/ItThinClientCriteriaQueryTest.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.runner.app.client;
+
+import static java.util.stream.Collectors.toList;
+import static org.apache.ignite.table.criteria.CriteriaQueryOptions.builder;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasSize;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.nio.file.Path;
+import java.util.Map;
+import org.apache.ignite.internal.testframework.WorkDirectory;
+import org.apache.ignite.table.RecordView;
+import org.apache.ignite.table.Tuple;
+import org.hamcrest.FeatureMatcher;
+import org.hamcrest.Matcher;
+import org.jetbrains.annotations.Nullable;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInfo;
+
+/**
+ * Thin client criteria query test.
+ */
+@SuppressWarnings("resource")
+public class ItThinClientCriteriaQueryTest extends ItAbstractThinClientTest {
+    /** {@inheritDoc} */
+    @Override
+    @BeforeAll
+    void beforeAll(TestInfo testInfo, @WorkDirectory Path workDir) throws InterruptedException {
+        super.beforeAll(testInfo, workDir);
+
+        populateData();
+    }
+
+    @Test
+    public void testBasicQueryCriteriaRecordBinaryView() {
+        var view = client().tables().table(TABLE_NAME).recordView();
+
+        var res = view.queryCriteria(null, null).stream().collect(toList());
+        assertThat(res, hasSize(3));
+
+        var ars = view.queryCriteriaAsync(null, null, builder().pageSize(2).build()).join();
+        assertEquals(2, ars.currentPageSize());
+    }
+
+    @Test
+    public void testBasicQueryCriteriaRecordPojoView() {
```

Comment:
I just see check sizes, but not content. Why?

Reply:
Added